flat_mutation_reader.cc

```diff
@@ -912,3 +915,47 @@ class generating_reader final : public flat_mutation_reader::impl {
 flat_mutation_reader make_generating_reader(schema_ptr s, std::function<future<mutation_fragment_opt> ()> get_next_fragment) {
     return make_flat_mutation_reader<generating_reader>(std::move(s), std::move(get_next_fragment));
 }
+
+bool mutation_fragment_stream_validator::operator()(const dht::decorated_key& dk) {
+    // FIXME: Validate fragment position monotonicity
+    //        https://github.com/scylladb/scylla/issues/4804
+    return true;
+}
+
+template<typename T>
+sstring position_desc(mutation_fragment::kind kind, const T& pos) {
+    return format("{}{}", kind == mutation_fragment::kind::range_tombstone ? "range_tombstone: " : "", pos);
```

Comment:
Why not print the `kind` unconditionally, using its natural `operator<<`?

It's not clear what T is. I think you want to accept `const position_in_partition_view&`.

Reply:
I'll try printing kind unconditionally as you suggested.
T could be either `position_in_partition` or position_in_partition_view.
The former can be converted to a view of course.